### PR TITLE
Update widget.py

### PR DESCRIPTION
**Pull request by Paradisee**
Add ValueError

### CODE REVIEW

File: kivy/uix/widget.py [hunk 1]
```diff
@@ -464,13 +464,15 @@ def add_widget(self, widget, index=0, canvas=None):
...
+            raise ValueError
+        elif index == 0 or len(self.children) == 0:
             self.children.insert(0, widget)
             canvas.add(widget.canvas)
         else:
             canvas = self.canvas
             children = self.children
-            if index >= len(children):
+            if index == len(children):
                 index = len(children)
```
Comment (denys-duchier) [inline kivy/uix/widget.py]: that seems redundant.

**Comment by Paradisee** [comment]
This is not consistent for me, because when add a widget to a Layout and set an index greater than the length of the children's list, you can't absolutely make it work! Instead you can Raise an IndexError. That's my point.

**Comment by denys-duchier** [comment]
I mean: look at lines 475,476

**Comment by matham** [comment]
I don't think this is a good idea. Why restrict it? What do we gain by raising an exception rather than by assuming that the user just wanted to stick it at the end? Also, it breaks backward compat.

Finally, python's list.insert allows indices that are larger than the list, so I don't think we should treat this any different. Just try `a = [1]`, and `a.insert(10, 2)` and you'll see it doesn't raise an exception.

**Comment by denys-duchier** [comment]
I agree.  I vote -1

**Comment by Paradisee** [comment]
Then i could put an index = 100000, where the list contains only 10 items. Coherence.

**Comment by matham** [comment]
@Paradisee yes, exactly, if you put 100,000 as index, the assumption is that you meant to just put it at the end of the list. That's really not a surprising assumption, especially given that that's how python list behaves. Anyway, it does break backward compat for no apparent benefit, so I'm closing for now.

**Pull request closed without merging**